keylime/cmd_exec.py

```diff
@@ -82,7 +82,7 @@ def run(
 
 # list_contains_substring checks whether a substring is contained in the given
 # list. The list may be the reterr from 'run' and contains bytes-like objects.
-def list_contains_substring(lst: List, substring: str) -> bool:
+def list_contains_substring(lst: List[bytes], substring: str) -> bool:
```

Comment:
The name, documentation and code suggest me `List[str]`. I propose to evaluate `List[str|bytes]`

Reply:
I changed the two `cast()`s for the return values of `_execute()` to be forced to be `bytes` via `assert`s. This includes `reterr` then that must be 'bytes' or maybe 'None' (not covered, yet). Let's see how this turns out after the test cases run.

Reply:
Should be good though:
```
import os, subprocess
proc=subprocess.Popen(['ls'], env=os.environ, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
out, err = proc.communicate()
type(err)
<class 'bytes'>
```